src/app/features/task-repeat-cfg/store/task-repeat-cfg.effects.ts

```diff
@@ -58,6 +66,131 @@ export class TaskRepeatCfgEffects {
+          const task = allTasks.find((aTask) => aTask.id === aAction.task.id)!;
+
+          if (task.repeatCfgId !== null) {
+            const repeatCfgForTask = await this._taskRepeatCfgService
+              .getTaskRepeatCfgById$(task.repeatCfgId)
+              .pipe(first())
+              .toPromise();
+
+            const taskChanges = aAction.task.changes;
+
+            // TODO: is there a better way to do this? Is there anything missing?
+            const repeatCfgChanges: Partial<TaskRepeatCfgCopy> = {
+              projectId: taskChanges.projectId ?? repeatCfgForTask.projectId,
+              title: taskChanges.title ?? repeatCfgForTask.title,
+              tagIds: taskChanges.tagIds ?? repeatCfgForTask.tagIds,
+              notes: taskChanges.notes ?? repeatCfgForTask.notes,
+            };
+
+            // TODO: Do we need to do this for all instances??
+            this._taskRepeatCfgService.updateTaskRepeatCfg(
+              task.repeatCfgId,
+              repeatCfgChanges,
+            );
+          }
+        }),
+      ),
+    { dispatch: false }, // Question: What exactly does this do?
+  );
+
+  /**
+   * When a main task is made repeatable, this function checks if there are subtasks.
+   * If that is the case, a repeat-cfg gets added for each subtask, too.
+   */
+  addTaskRepeatCfgForSubTasksOf: any = createEffect(
+    () =>
+      this._actions$.pipe(
+        ofType(addTaskRepeatCfgToTask),
+        tap(async (aAction) => {
+          // TODO: is there an easier way to get to the parent task?
+          const allTasks = await this._taskService.allTasks$.pipe(first()).toPromise();
+          const parentTask = allTasks.find((aTask) => aTask.id === aAction.taskId);
+          const parentTaskRepeatCfg = aAction.taskRepeatCfg;
+
+          if (parentTask !== undefined && parentTask.subTaskIds.length > 0) {
+            for (const aSubTaskId of parentTask.subTaskIds) {
+              const task = allTasks.find((aTask) => aTask.id === aSubTaskId)!;
+
+              const repeatCfg = {
+                ...parentTaskRepeatCfg,
+                // TODO: anything missing in this list that should not be overwritten by the parent?
+                title: task.title,
+                notes: task.notes,
+                defaultEstimate: task.timeEstimate, // is this correct?
+                parentId: parentTask.repeatCfgId,
+              };
+
+              this._taskRepeatCfgService.addTaskRepeatCfgToTask(
+                task.id,
+                task.projectId,
+                repeatCfg,
+              );
+            }
+          }
+        }),
+      ),
+    { dispatch: false }, // Question: What exactly does this do?
+  );
+
+  /**
+   * When adding a sub task, this function checks if the parent is a repeatable task and therefore the sub-task also has to be.
+   * If that is the case, a repeat-cfg gets added for each subtask, too.
+   */
+  addTaskRepeatCfgForSubTask: any = createEffect(
```

Comment:
Imho subtasks should not be repeatable tasks, but just subtasks of a task with a repeatable config.

Reply:
Since finished tasks get deleted (at least that's what it seemed like when I tried to observe how the app works?) and I am not confident enough at this point to try and change things on such a fundamental level, I based this approach on the fact that repeatCfgs are persisted and can be used to store the information of the sub tasks. I basically mirrored that structure with the repeatCfs:

All Tasks (sub + parent) are stored as tasks (all together) in the store (only difference is if they have subtask-ids or a parent-Id or not)
All repeatCfgs (sub + parent) are stored as repeatCfs (all together) in the store (only difference is that they have a parent id or not)

Since all information of a repeatable task gets generated from a repeatable config, I only see 2 ways of storing the information for the sub tasks:
- storing all information about all subtasks in the parent repeat config (which would go against the structure that is currently implemented for normal tasks and therefore would feel like bad practise to me, but that might just be my view)
- storing the information about sub tasks as a subtask-repeatable-config

Definitely let me know if I missed something.

Reply:
Afaics subtasks don't need to "know" that they are a child task of a repeatable task and I also think different parent repeatable tasks should be allowed to have different sub tasks. To solve this I would not reference "real" sub tasks but rather provide a simple input that allows to create a list of sub tasks with no more information than maybe the title. 

Using real "subtasks" as reference point creates all sorts of messy problems, when moving them around or deleting them or their parent.